k2/python/k2/fsa_algo.py

```diff
@@ -464,11 +464,8 @@ def connect(fsa: Fsa) -> Fsa:
             fsa.properties & fsa_properties.COACCESSIBLE != 0:
         return fsa
 
-    assert fsa.is_cpu()
```

Comment:
Please also delete line 446 and update k2/python/tests/connect_test.py to test for CUDA.